frontend/src/metabase/components/EmptyState.jsx

```diff
@@ -1,35 +1,52 @@
+/* @flow */
 import React from "react";
-import PropTypes from "prop-types";
-import { Link } from "react-router";
+import {Link} from "react-router";
+import cx from "classnames";
+/*
+ * EmptyState is a component that can
+ *   1) introduce a new section of Metabase to a user, and encourages the user to take an action
+ *  2) indicate an empty result after a user-triggered search/query
+ */
 
 import Icon from "metabase/components/Icon.jsx";
 
-const EmptyState = ({ title, message, icon, image, action, link }) =>
-    <div className="text-centered text-brand-light my2">
+type EmptyStateProps = {
+    message: (string | React$Element<any>),
+    title?: string,
+    icon?: string,
+    image?: string,
+    imageClassName?: string,
+    action?: string,
+    link?: string,
+    onActionClick?: () => void,
+    smallDescription?: boolean
+}
+
+const EmptyState = ({title, message, icon, image, imageClassName, action, link, onActionClick, smallDescription = false}: EmptyStateProps) =>
+    <div className="text-centered text-brand-light my2" style={smallDescription ? {} : {width: "350px"}}>
         { title &&
-            <h2 className="text-brand mb4">{title}</h2>
+        <h2 className="text-brand mb4">{title}</h2>
```

Comment:
Usually we indent here (prettier does as well) but since we're planning on switching to prettier soon I'm not too concerned about style.